### PR TITLE
chore: upgrade node version to 16

### LINKED ISSUE #3556: Update node to 16
We are currently on Node 14, which is in maintenance mode and nearing its end of life on 2023-04-30.
We should upgrade to Node 16, which is also in maintenance mode with an end of life of 2023-09-11.

We should _also_ upgrade to Node 18 (EOL 2025-04-30), but let's do it in a separate issue and PR.

### PR DESCRIPTION
**Pull request by Walther**
**What this PR does / why we need it**: Upgrades the `node` version from `14` to `16`

- `node` upgrade from `14` to `16`
- `better-sqlite3` upgrade from `7.5.0` to `7.5.3`
- `@garden-io/platform-api-types"` from `1.271.0` to `1.345.0` 
- `shared-machine-config` from `ubuntu-2004:202010-01` to `ubuntu-2204:2022.10.2`
- `microk8s` group fixes, removing the need to use `sudo` in the `microk8s` tests
-   `resource_class` on `microk8s` tests increased from `medium` to `large` (this can be reverted if we want)

Fixes #3556 

Note: there are experimental commits inbetween, so please review just the final result. I will squash-merge this.

### CODE REVIEW

File: cli/src/build-pkg.ts [hunk 1]
```diff
@@ -48,10 +48,10 @@ interface TargetSpec {
 }
 
 const targets: { [name: string]: TargetSpec } = {
-  "macos-amd64": { pkgType: "node14-macos-x64", handler: pkgMacos, nodeBinaryPlatform: "darwin" },
-  "linux-amd64": { pkgType: "node14-linux-x64", handler: pkgLinux, nodeBinaryPlatform: "linux" },
-  "windows-amd64": { pkgType: "node14-win-x64", handler: pkgWindows, nodeBinaryPlatform: "win32" },
-  "alpine-amd64": { pkgType: "node14-alpine-x64", handler: pkgAlpine, nodeBinaryPlatform: "linuxmusl" },
+  "macos-amd64": { pkgType: "node16-macos-x64", handler: pkgMacos, nodeBinaryPlatform: "darwin" },
```
Comment (stefreak) [inline cli/src/build-pkg.ts:51]: can we add `macos-arm64` with node16?

Comment (Walther) [reply cli/src/build-pkg.ts:51]: i'd love to support arm for both mac as well as linux targets!
let's keep that in a separate pr though, as there are some extra requirements to conform to + possibly extra testing to set up

**Comment by edvald** [comment]
@Walther I'd love to get this back in motion, since we're running into some issues with being stuck on old packages (mainly due to ESM module support). 

Looks like much of the work is done. Would it make sense if I pick this up or would you like to wrap it up?

**Comment by Walther** [comment]
Happy to pick this up!

**Comment by Walther** [comment]
Closing in favor of a new PR with a fresher branch.

**Comment by Walther** [comment]
Not sure what is happening with the `vm-1.24-microk8s` test - failed consistently through 5 retries. Ideas or help appreciated! 🙇‍♂️

**Review by stefreak — DISMISSED** [review]
Great work! ❤️

Once the tests pass I think it's a good idea to merge this to main and then port to 0.13 so we have the latest node version in both 0.12 and 0.13 as we'll have to keep supporting 0.12 for some time.

**Comment by Walther** [comment]
That is the plan indeed!
- when we get tests to pass, merge this to main
- port to `0.13` branch
- land the node 16 update PR on platform repository as well
- repeat full process for node 18

**Review by stefreak — APPROVED** [review]
Wow! Good job figuring out these nasty circleci microk8s issues 😨 

🎉

**Pull request merged**